src/main.jsx

```diff
@@ -19,6 +20,7 @@ const router = createBrowserRouter(
     <Route path="/" element={<Layout />}>
       <Route index path="/" element={<Home />} />
       <Route path="/login" element={<Login />} />
+      <Route path="/me" element={<Profile />} />
```

Comment:
please make this `Route` as `Protected`, so that without Logging in it can not be accessed.

Example:

Creating protected routes in React Router version 6 involves setting up a mechanism to restrict access to certain routes based on user authentication or other criteria. Here's a general approach to creating protected routes:

1. Setup Authentication Context
First, you'll need some form of authentication context to keep track of the user's authentication status. This can be done using React's Context API.

AuthContext.js
```
import React, { createContext, useState } from 'react';

export const AuthContext = createContext();

export const AuthProvider = ({ children }) => {
    const [isAuthenticated, setIsAuthenticated] = useState(false);

    // Add your authentication logic here

    return (
        <AuthContext.Provider value={{ isAuthenticated, setIsAuthenticated }}>
            {children}
        </AuthContext.Provider>
    );
};
```
2. Create a Protected Route Component
Next, create a component that wraps around the routes you want to protect. This component will check the authentication status and either render the component or redirect the user.

ProtectedRoute.js
```
import React, { useContext } from 'react';
import { Navigate, useLocation } from 'react-router-dom';
import { AuthContext } from './AuthContext';

export const ProtectedRoute = ({ children }) => {
    const { isAuthenticated } = useContext(AuthContext);
    const location = useLocation();

    if (!isAuthenticated) {
        // Redirect them to the login page, but save the current location they were trying to go to
        return <Navigate to="/login" state={{ from: location }} replace />;
    }

    return children;
};
```
3. Wrap Your Routes
In your routing configuration, wrap the routes you want to protect with the ProtectedRoute component.

App.js
```
import React from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { ProtectedRoute } from './ProtectedRoute';
import { AuthProvider } from './AuthContext';
import Home from './Home';
import Login from './Login';
import Dashboard from './Dashboard';

const App = () => {
    return (
        <Router>
            <AuthProvider>
                <Routes>
                    <Route path="/" element={<Home />} />
                    <Route path="/login" element={<Login />} />
                    <Route path="/dashboard" element={
                        <ProtectedRoute>
                            <Dashboard />
                        </ProtectedRoute>
                    } />
                    {/* Add more protected routes here */}
                </Routes>
            </AuthProvider>
        </Router>
    );
};

export default App;
```
In this setup:

AuthContext provides the authentication state across the application.
ProtectedRoute checks the authentication status and redirects unauthenticated users to the login page.
In App.js, the routes are wrapped with ProtectedRoute to ensure they are accessible only by authenticated users.